File: backend/src/main/java/com/shridhar/prescripto/backend/model/Prescription.java
package com.shridhar.prescripto.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Prescription {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private User doctor;

    @ManyToOne
    private User patient;

    private LocalDateTime dateIssued;

    @OneToMany(mappedBy = "prescription", cascade = CascadeType.ALL)
    private List<Medication> medications;
}

File: android/app/src/main/java/com/shridhar/prescripto/model/Prescription.kt
package com.shridhar.prescripto.model

import java.util.UUID

data class Prescription(
    val id: String = UUID.randomUUID().toString(),
    val patientId: String,
    val doctorId: String,
    val doctorName: String,
    val patientName: String,
    val dateIssued: Long,
    val medications: List<Medication>,
    val notes: String = "",
    val followUpDate: Long? = null
)
